cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -196,38 +517,683 @@ impl BuildEnv for BuildEnvNix {
+        let runtime = &result.runtime;
+        assert!(runtime.join("activate.d/bash").exists());
+        assert!(runtime.join("activate.d/zsh").exists());
+        assert!(runtime.join("etc/profile.d").is_dir());
+
+        let develop = &result.develop;
+        assert!(develop.join("activate.d/bash").exists());
+        assert!(develop.join("activate.d/zsh").exists());
+        assert!(develop.join("etc/profile.d").is_dir());
+    }
+
+    #[test]
+    fn build_contains_lockfile() {
+        let result = &*BUILDENV_RESULT_SIMPLE_PACKAGE;
+        let runtime = &result.runtime;
+        assert!(runtime.join("manifest.lock").exists());
+
+        let develop = &result.develop;
+        assert!(develop.join("manifest.lock").exists());
+    }
+    #[test]
+    fn build_contains_build_script_and_output() {
+        let buildenv = BuildEnvNix;
+        let lockfile_path = GENERATED_DATA.join("envs/build-noop/manifest.lock");
+        let result = buildenv.build(&lockfile_path, None).unwrap();
+
+        let runtime = result.runtime.as_ref();
+        let develop = result.develop.as_ref();
+        let build_hello = result.manifest_build_runtimes.get("build-hello").unwrap();
+
+        assert!(runtime.join("package-builds.d/hello").exists());
+        assert!(develop.join("package-builds.d/hello").exists());
+        assert!(build_hello.join("package-builds.d/hello").exists());
+    }
+
+    #[test]
+    fn build_on_activate_lockfile() {
+        let buildenv = BuildEnvNix;
+        let lockfile_path = MANUALLY_GENERATED.join("buildenv/lockfiles/on-activate/manifest.lock");
+        let result = buildenv.build(&lockfile_path, None).unwrap();
+
+        let runtime = &result.runtime;
+        assert!(runtime.join("activate.d/hook-on-activate").exists());
+
+        let develop = &result.develop;
+        assert!(develop.join("activate.d/hook-on-activate").exists());
+    }
+
+    #[test]
+    fn verify_contents_of_requisites_txt() {
+        let result = &*BUILDENV_RESULT_SIMPLE_PACKAGE;
+
+        let runtime = result.runtime.as_ref();
+        let develop = result.develop.as_ref();
+
+        for out_path in [runtime, develop] {
+            let requisites_path = out_path.join("requisites.txt");
+            assert!(requisites_path.exists());
+
+            let requisites: HashSet<String> = std::fs::read_to_string(&requisites_path)
+                .unwrap()
+                .lines()
+                .map(String::from)
+                .collect();
+
+            let output = Command::new("nix-store")
+                .arg("-qR")
+                .arg(out_path)
+                .output()
+                .expect("failed to execute process");
+
+            assert!(output.status.success());
+
+            let store_paths: HashSet<String> = String::from_utf8_lossy(&output.stdout)
+                .lines()
+                .map(String::from)
+                .collect();
+
+            assert_eq!(requisites, store_paths);
```

Comment:
(nb) I didn't understand this comment from the original test. Is there anything to follow-up on?

https://github.com/flox/flox/blob/6916afa816b6b59725a735f59f70b7cb643036f0/cli/tests/buildenv.bats#L198-L203

Reply:
I dont quite understand that either, maybe @limeytexan can chime in to explain?